hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RegionCoprocessorHost.java

```diff
@@ -299,6 +304,11 @@ public RegionCoprocessorHost(final HRegion region,
           } catch (NoSuchMethodException ignore) {
           }
           clazz = clazz.getSuperclass();
+
```

Comment:
Actually change can be at begin of this loop
if (clazz == null) {
            // we must have directly implemented RegionObserver
            hasCustomPostScannerFilterRow = true;
            break out;
          }
In 1.x we have interface with no def impl. So we have to consider like hasCustomPostScannerFilterRow = true then. But in 2.x we have Interface with default impl.  So its safe to change there to be hasCustomPostScannerFilterRow = false only.
In fact that check can be as urs
if (clazz == Object.class) {
break out;
}

Reply:
> In fact that check can be as urs
> if (clazz == Object.class) {
> break out;
> }

make sense, will update the PR.